### PR TITLE
media-gfx/nomacs: 3.19.1 version bump with Qt6 porting

### PR DESCRIPTION
**Pull request by band-a-prend**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by band-a-prend** [comment]
@novomesk 
I tried to port/adapt nomacs-plugins to Qt6. It's built and seems to be working.
Could you please review `files/nomacs-plugins-3.16_use_Qt6.patch`?

**Comment by a17r** [comment]
The real question is: Is it pending or, better, submitted upstream?

**Comment by band-a-prend** [comment]
@a17r the nomacs-plugins are seems being unmantained there is no activity for 4 years and some other packagers use own patches for qt6 porting. At least after I prepared own patch I found some patches from arch aur and openmandriva.

The nomacs itself already compatible with qt6 and as @novomesk did latest commits to nomacs upstream (but not to nomacs-plugins repo) I asked to check my nomacs-plugins patch.

**Comment by novomesk** [comment]
The repo for nomacs plugins changed.

The Windows build version now uses https://github.com/novomesk/nomacs-plugins/tree/3.19.0

**Comment by band-a-prend** [comment]
@novomesk 
Great! I then will make changes to use tagged tarball from your updated repo and then will rebase.

**Comment by band-a-prend** [comment]
Done. 
The updated upstream `nomacs-plugins` tagged repo is used now and no additional patches are required for Qt6.

I'm not sure is the SLOT'ing required for media-gfx/nomacs? I.e. should I change `SLOT="0/5"` for `media-gfx/nomacs-3.16.224-r6` and `SLOT="0/6"` for `media-gfx/nomacs-3.19.1`?

**Comment by novomesk** [comment]
I think slot change is not needed/required here. But that's just my opinion.

File: media-gfx/nomacs/nomacs-3.19.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		mv -v ../${PLUGIN_PKG} ImageLounge/plugins || die
+	fi
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DQT_VERSION_MAJOR=6
+		-DQT_QMAKE_EXECUTABLE=qmake6 # bug 847112
```
Comment (juippis) [inline media-gfx/nomacs/nomacs-3.19.1.ebuild:58]:
```
 * QA Notice: One or more CMake variables were not used by the project:
 *   QT_QMAKE_EXECUTABLE
```

**Review by juippis — COMMENTED** [review]
Also the test phase is a bit sad here, 1 out of 1 tests fail. If you could take a look it'd be awesome, but mostly I'm just making sure the CMAke variable isn't something critical.

File: media-gfx/nomacs/nomacs-3.19.1.ebuild (hunk 1 above)
Comment (band-a-prend) [reply media-gfx/nomacs/nomacs-3.19.1.ebuild]: I done the following changes to enable run tests. But I could disable it if needed:
```patch
@@ -16,7 +16,8 @@ CMAKE_USE_DIR="${S}/ImageLounge"
 LICENSE="GPL-3+"
 SLOT="0"
 KEYWORDS="~amd64 ~arm64 ~riscv ~x86 ~amd64-linux"
-IUSE="+opencv plugins raw +tiff zip"
+IUSE="+opencv plugins raw +tiff test zip"
+RESTRICT="!test? ( test )"
 
 REQUIRED_USE="
        raw? ( opencv )
@@ -35,7 +36,10 @@ RDEPEND="
        )
        zip? ( dev-libs/quazip:0=[qt6(+)] )
 "
-DEPEND="${RDEPEND}"
+DEPEND="
+       ${RDEPEND}
+       test? ( dev-cpp/gtest )
+"
 BDEPEND="
        dev-qt/qttools:6[linguist]
        virtual/pkgconfig
@@ -55,7 +59,6 @@ src_prepare() {
 src_configure() {
        local mycmakeargs=(
                -DQT_VERSION_MAJOR=6
-               -DQT_QMAKE_EXECUTABLE=qmake6 # bug 847112
                -DENABLE_CODE_COV=OFF
                -DUSE_SYSTEM_QUAZIP=ON
                -DENABLE_TRANSLATIONS=ON
@@ -63,7 +66,13 @@ src_configure() {
                -DENABLE_PLUGINS=$(usex plugins)
                -DENABLE_RAW=$(usex raw)
+               -DENABLE_TESTING=$(usex test)
                -DENABLE_TIFF=$(usex tiff)
                -DENABLE_QUAZIP=$(usex zip)
        )
        cmake_src_configure
 }
+
+src_test() {
+       cmake_build core_tests
+       cmake_src_test
+}
```

**Comment by band-a-prend** [comment]
Ping!

**Review by juippis — APPROVED** [review]
Sorry, and thanks!

**Pull request closed without merging**